gems/pending/middleware/events/middleware_event_monitor.rb

```diff
@@ -0,0 +1,31 @@
+class MiddlewareEventMonitor
+  def initialize(ems, start_time)
+    @ems = ems
+    @alerts_client = ems.alerts_connect
+    # TODO: can we store and resume where we left off?  To avoid duplicates, only look for events now or later,
+    # so basically we miss events when we're not running...
+    @start_time = start_time || Time.now.to_i * 1000
+  end
+
+  def start
+    # nothing to do, not an active connection, just making rest calls
+  end
+
+  def stop
+    # nothing to do, not an active connection, just making rest calls
+  end
+
+  def each
+    # Just fetch up to the current time, then rest current time as the next start time
+    end_time = Time.now.to_i * 1000
+    $mw_log.debug("Catching Events for interval [Start=#{@start_time}, End=#{end_time}]")
+
+    @alerts_client.list_events("startTime" => @start_time, "endTime" => end_time).each do |event|
+      yield event
```

Comment:
I think this does not work as you expect.
The block will only be executed for the events that are returned by `list_events` and then the method returns.
And the calling loop will also terminate.

Have a look at the pattern from @bronaghs which also deals with REST calls

https://github.com/ManageIQ/manageiq/pull/7439/files#diff-be7f5bc4284aee5245a4bf6c7ec4b9f9R29
https://github.com/ManageIQ/manageiq/pull/7439/files#diff-703f0a31b282a0170102b0ff38cc68edR23


@pilhuhn - side note: does Hawkular require a startTime and endTime? Or can the results be paged and the request asks only for a startTime?